src/renderer/containers/App/index.tsx

```diff
@@ -115,11 +117,17 @@ export default function App() {
       // the addition of the fileId i.e. FSS's completion, has failed,
       // or requires this clients intervention to retry
       if (
-        job.service === Service.FILE_STORAGE_SERVICE &&
-        (job.serviceFields?.fileId || job.status === JSSJobStatus.FAILED || job.currentStage === UploadStatus.RETRY)
-      ) {
-        dispatch(receiveFSSJobCompletionUpdate(job as FSSUpload));
-      } else if (job.serviceFields?.type === "upload") {
+        job.service === Service.FILE_STORAGE_SERVICE) {
+          if (job.serviceFields?.fileId || job.status === JSSJobStatus.FAILED || job.currentStage === UploadStatus.RETRY) {
+            dispatch(receiveFSSJobCompletionUpdate(job as FSSUpload));
+          } else if(job.serviceFields?.preUploadMd5 && (job.serviceFields.preUploadMd5 !== job.serviceFields?.fileSize)) {
+            dispatch(updateUploadProgressInfo(job.jobId, { bytesUploaded: job.serviceFields?.preUploadMd5, totalBytes: job.serviceFields?.fileSize, step: Step.ONE }));
+          } else if(job.serviceFields?.currentFileSize && (job.serviceFields.currentFileSize!== job.serviceFields?.fileSize)) {
+            dispatch(updateUploadProgressInfo(job.jobId, { bytesUploaded: job.serviceFields?.currentFileSize, totalBytes: job.serviceFields?.fileSize, step: Step.TWO }));
```

Comment:
This works with the non-shortcut upload too right?

Reply:
right